ts/client/src/client.ts

```diff
@@ -964,22 +960,21 @@ export class MangoClient {
         mangoAccount,
         perpMarketIndex,
       );
-      healthAccountsToExclude.push(perpMarket.publicKey, perpMarket.oracle);
+      perpMarketsToExclude.push(perpMarket);
       instructions.push(deactivatingPositionIx);
     }
 
     for (const index in mangoAccount.tokensActive()) {
       const indexNum = Number(index);
-      const accountsToExclude = [...healthAccountsToExclude];
       const token = mangoAccount.tokensActive()[indexNum];
       const bank = group.getFirstBankByTokenIndex(token.tokenIndex);
-      //to withdraw from all token accounts we need to exclude previous tokens pubkeys
-      //used to build health remaining accounts
+      // To withdraw from a particular token account we need to exclude previously
+      // withdrawn tokens
       if (indexNum !== 0) {
         for (let i = indexNum; i--; i >= 0) {
           const prevToken = mangoAccount.tokensActive()[i];
           const prevBank = group.getFirstBankByTokenIndex(prevToken.tokenIndex);
-          accountsToExclude.push(prevBank.publicKey, prevBank.oracle);
+          banksToExclude.push(prevBank);
```

Comment:
Since this is in the loop over token accounts, won't this add the prevBanks again and again? Couldn't we add to banksToExclude every time a withdraw ix was made?